pkg/cloudprovider/huaweicloud/huaweicloud.go

```diff
@@ -767,13 +769,34 @@ func (h *CloudProvider) listenerDeploy() error {
 		invalidServiceCache: gocache.New(5*time.Minute, 10*time.Minute),
 	}
 
+	// init security group listener
+	cloudConfigSecId := h.cloudConfig.VpcOpts.SecurityGroupID
+	if len(cloudConfigSecId) == 0 {
+		return nil
+	}
```

Comment:
This is a dangerous operation, and many functions will be unavailable after return.